### PR TITLE
Add contribution guidelines for Taskfiles.

### PR DESCRIPTION
**Pull request by kirkrodrigues**
# Description

This PR starts the contribution guidelines section and open-sources our guidelines for Taskfiles.

# Validation performed

* `task serve`
* Validated the new docs look correct.

### CODE REVIEW

File: docs/dev-guide/contrib-guides-taskfiles.md [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+## Paths in commands
+
+All paths used in commands should be quoted.
+
+**Example**
+
+```yaml
+my-task:
+  cmds:
+    - "touch '{{.FILE_PATH}}'"
```
Comment (junhaoliao) [inline docs/dev-guide/contrib-guides-taskfiles.md:33]: I assume we don't mention "strings should be quoted" in the first place because "have automated linting processes", right?

File: docs/dev-guide/contrib-guides-taskfiles.md [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+## Built-in variables
+
+1. Don't use the variable `{{.ROOT_TASKFILE}}`
+   * v3.35.1 has a bug that makes it equivalent to `ROOT_DIR` (i.e., the parent directory of the root
+     Taskfile, rather than the path to the root Taskfile).
+2. When using the variable `{{.TASK}}` in a task within a namespace (e.g., it's in a nested
+   Taskfile), replace the `:` in the name with `#` as follows: `{{.TASK | replace \":\" \"#\"}}`.
+   * This ensures the task name can be used as a filename.
+
+## `sources` and `generates`
```
Comment (junhaoliao) [inline docs/dev-guide/contrib-guides-taskfiles.md:47]: Do we want to mention how (and why) we want to checksum certain tasks instead of this `sources` + `generates` pattern?
If we're short of time, we can do that in another PR.

File: docs/dev-guide/contrib-guides-taskfiles.md [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+* Execution control (ordered by execution order)
+  * `prompt`
+  * `run`
+  * `deps`
+  * `preconditions`
+  * `status`
+  * `cmds`
+  * `ignore_error`
+  * `interactive`
+* Logging
```
Comment (junhaoliao) [inline docs/dev-guide/contrib-guides-taskfiles.md:128]: I feel it might be more intuitive to place these next to `Descriptors`.

File: docs/dev-guide/contrib-guides-taskfiles.md [hunk 4]
```diff
@@ -0,0 +1,161 @@
+# Taskfiles
```
Comment (junhaoliao) [inline docs/dev-guide/contrib-guides-taskfiles.md:1]: Although not Taskfile-specific, do we want to mention in YAML how we place array elements into separate lines where they can't fit into one?

File: docs/dev-guide/contrib-guides-taskfiles.md [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+  * `ignore_error`
+  * `interactive`
+* Logging
+  * `label`
+  * `prefix`
+  * `silent`
+
+## Ordering of list/map entries
+
+List/map entries should be ordered alphabetically (except for `vars`). Where there are several
```
Comment (junhaoliao) [inline docs/dev-guide/contrib-guides-taskfiles.md:135]: Do we want to mention the alphabetical order is case insensitive? The leading characters might be in different cases in paths specified in `sources` and `generates` for example.

File: docs/dev-guide/contrib-guides-taskfiles.md (hunk 4 above)
Comment (kirkrodrigues) [reply docs/dev-guide/contrib-guides-taskfiles.md:1]: Good point. I would suggest we do it in another PR when we add YAML guidelines, and then add a link to them from the Taskfile guidelines.

File: docs/dev-guide/contrib-guides-taskfiles.md (hunk 2 above)
Comment (kirkrodrigues) [reply docs/dev-guide/contrib-guides-taskfiles.md:47]: Good point. Added.

File: docs/dev-guide/contrib-guides-taskfiles.md (hunk 1 above)
Comment (kirkrodrigues) [reply docs/dev-guide/contrib-guides-taskfiles.md:33]: Yep

File: docs/dev-guide/contrib-guides-taskfiles.md (hunk 5 above)
Comment (kirkrodrigues) [reply docs/dev-guide/contrib-guides-taskfiles.md]: True, added.

**Review by junhaoliao — APPROVED** [review]
The PR title is fine for the squashed commit message.

File: docs/dev-guide/contrib-guides-taskfiles.md [hunk 6]
```diff
@@ -0,0 +1,282 @@
...
+    - "touch '{{.FILE_PATH}}'"
+    - |-
+      echo "Hello, world" > "{{.FILE_PATH}}"
+```
+
+## Built-in variables
+
+1. Don't use the variable `{{.ROOT_TASKFILE}}`
+   * v3.35.1 has a bug that makes it equivalent to `ROOT_DIR` (i.e., the parent directory of the root
+     Taskfile, rather than the path to the root Taskfile).
```
Comment (junhaoliao) [inline docs/dev-guide/contrib-guides-taskfiles.md:42]: this should be removed now given these days our projects use a newer version of go-task which doesn't have this bug

@coderabbitai create an issue to track
